### ISSUE TITLE
ReRanker - You don't have the necessary dependencies installed ... 

### ISSUE THREAD

**Issue by geg00** [AnswerDotAI/rerankers#29]
python: 3.11.6
Windows 11 machine

I've installed 
```
!pip install "rerankers[all]
```
deployment_name = "gpt-4o"
ranker = Reranker(f"azure/{deployment_name}", model_type="rankllm", api_key = OPENAI_API_KEY)

Loading RankLLMRanker model azure/gpt-4o
You don't have the necessary dependencies installed to use RankLLMRanker.
Please install the necessary dependencies for RankLLMRanker by running `pip install "rerankers[rankllm]"` or `pip install "rerankers[all]" to install the dependencies for all reranker types.

```
if ranker is not None:
    results = ranker.rank(query="I love you", docs=["I hate you", "I really like you"], doc_ids=[0,1])
else:
    print("Ranker object is None.")

>> Ranker object is None
```

ranker = Reranker('flashrank')

Loading default flashrank model for language en
Default Model: ms-marco-MiniLM-L-12-v2
Loading FlashRankRanker model ms-marco-MiniLM-L-12-v2
You don't have the necessary dependencies installed to use FlashRankRanker.
Please install the necessary dependencies for FlashRankRanker by running `pip install "rerankers[flashrank]"` or `pip install "rerankers[all]" to install the dependencies for all reranker types.

Any help?

**Comment by geg00** [comment]
This one works 
```
ranker = Reranker('cross-encoder')

if ranker is not None:
    results = ranker.rank(query="I love you", docs=["I hate you", "I really like you"], doc_ids=[0,1])
else:
    print("Ranker object is None.")

results

```

**Comment by geg00** [comment]
Requirement already satisfied: **torch** in [c:\users\hbb9279\repos\azure-functions\.venv\lib\site-packages](file:///C:/users/hbb9279/repos/azure-functions/.venv/lib/site-packages) (2.4.1)
Requirement already satisfied: torchvision in [c:\users\hbb9279\repos\azure-functions\.venv\lib\site-packages](file:///C:/users/hbb9279/repos/azure-functions/.venv/lib/site-packages) (0.18.1+cpu)
Collecting torchvision
  Obtaining dependency information for torchvision from https://files.pythonhosted.org/packages/f8/69/dc769cf54df8e828c0b8957b4521f35178f5bd4cc5b8fbe8a37ffd89a27c/torchvision-0.19.1-cp311-cp311-win_amd64.whl.metadata
  Using cached torchvision-0.19.1-cp311-cp311-win_amd64.whl.metadata (6.1 kB)
Requirement already satisfied: torchaudio in [c:\users\hbb9279\repos\azure-functions\.venv\lib\site-packages]...
Collecting protobuf (from rerankers[all])
...
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?0ccb6951-95e8-4b49-aafb-5971bafbb9db) or open in a [text editor](command:workbench.action.openLargeOutput?0ccb6951-95e8-4b49-aafb-5971bafbb9db). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "C:\Users\hbb9279\Repos\azure-functions\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "C:\Users\hbb9279\Repos\azure-functions\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\hbb9279\Repos\azure-functions\.venv\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\hbb9279\AppData\Local\Temp\pip-build-env-a1obzoox\overlay\Lib\site-packages\setuptools\build_meta.py", line 332, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "C:\Users\hbb9279\AppData\Local\Temp\pip-build-env-a1obzoox\overlay\Lib\site-packages\setuptools\build_meta.py", line 302, in _get_build_requires
          self.run_setup()
        File "C:\Users\hbb9279\AppData\Local\Temp\pip-build-env-a1obzoox\overlay\Lib\site-packages\setuptools\build_meta.py", line 503, in run_setup
          super().run_setup(setup_script=setup_script)
        File "C:\Users\hbb9279\AppData\Local\Temp\pip-build-env-a1obzoox\overlay\Lib\site-packages\setuptools\build_meta.py", line 318, in run_setup
          exec(code, locals())
        File "<string>", line 21, in <module>
      **ModuleNotFoundError: No module named 'torch'**
...
note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.2.1 -> 24.2
[notice] To update, run: python.exe -m pip install --upgrade pip
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?0a17f9d1-7d4d-413c-9f9a-3ae7a841e6f0) or open in a [text editor](command:workbench.action.openLargeOutput?0a17f9d1-7d4d-413c-9f9a-3ae7a841e6f0). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

**Comment by bclavie** [comment]
This issue isn’t directly related to rerankers, it seems like your environment isn’t resolving properly.

Would you be able to post an untrucated error trace to try and figure out what exactly is causing the issue?

**Comment by hoorangyee** [comment]
I believe this issue stems from rerankers itself. 

The problem occurs because the import paths for vicuna_reranker and zephyr_reranker in [models/rankllm_reranker.py](https://github.com/AnswerDotAI/rerankers/blob/main/rerankers/models/rankllm_ranker.py) are not aligned with the latest version of rank_llm. 

This causes the error when attempting to use rankllm within rerankers. I've submitted a fix for this in this [PR](https://github.com/AnswerDotAI/rerankers/pull/57).

**Comment by bclavie** [comment]
> I believe this issue stems from rerankers itself.
> 
> The problem occurs because the import paths for vicuna_reranker and zephyr_reranker in [models/rankllm_reranker.py](https://github.com/AnswerDotAI/rerankers/blob/main/rerankers/models/rankllm_ranker.py) are not aligned with the latest version of rank_llm.
> 
> This causes the error when attempting to use rankllm within rerankers. I've submitted a fix for this in this [PR](https://github.com/AnswerDotAI/rerankers/pull/57).

Thank you for the PR! I think the issue is not fully related (this specific issue was about `torch` not being installed, which is a broader problem), but yours does fix the other, definitely rerankers-related issue of rankllm not importing! The fix is live as of 0.7.1 on pypi.

**Comment by OverSamu** [comment]
This error occurs again.

Python 3.12.3
rerankers 0.10.0
Windows 10

`pip install "rerankers[rankllm]"`

```
Loading RankLLMRanker model gpt-4o (this message can be suppressed by setting verbose=0)
You don't have the necessary dependencies installed to use RankLLMRanker.
Please install the necessary dependencies for RankLLMRanker by running `pip install "rerankers[rankllm]"` or `pip install "rerankers[all]"` to install the dependencies for all reranker types.
```
